internal/mark.go

```diff
@@ -0,0 +1,16 @@
+package internal
+
+import (
+	"github.com/wal-g/storages/storage"
+	"github.com/wal-g/wal-g/utility"
+)
+
+type MarkHandler struct {
+	Mark func(uploader *Uploader, folder storage.Folder, backupName string, toPermanent bool)
+}
+
+func (m *MarkHandler) HandleBackupMark(uploader *Uploader, backupName string, toPermanent bool) {
+	folder := uploader.UploadingFolder
```

Comment:
Most of the MarkHandler logic seems to be done by external Mark function, maybe consider different decomposition?